OnProfileUpdate/__tests__/handler.test.ts

```diff
@@ -0,0 +1,210 @@
+  },
+  {
+    email: "Reed_Klocko@example.com",
+    fiscalCode: "PVQEBX22A89Y092X",
+    isEmailValidated: true,
+    isInboxEnabled: false,
+    id: "PVQEBX22A89Y092X-0000000000000000",
+    version: 0,
+    _self: "ad893263-21a5-43af-856b-88bc80fdb5a2"
+  },
+  {
+    email: "derd@example.com",
+    fiscalCode: "PVQEBX22A89Y092X",
+    isEmailValidated: false,
+    isInboxEnabled: false,
+    acceptedTosVersion: 1,
+    id: "PVQEBX22A89Y092X-0000000000000001",
+    version: 1,
+    _self: "19e3eeb9-0fc0-472b-8df9-b29eab5a2d50"
+  },
+  {
+    email: "derd@example.com",
+    fiscalCode: "PVQEBX22A89Y092X",
+    isEmailValidated: true,
+    isInboxEnabled: false,
+    acceptedTosVersion: 2,
+    id: "PVQEBX22A89Y092X-0000000000000002",
+    version: 2,
+    _self: "4b4c94d4-a350-4f27-9c76-ba669eca48a9"
+  },
+  {
+    email: "cittadinanzadigitale@teamdigitale.governo.it",
+    fiscalCode: "ISPXNB32R82Y766D",
+    isEmailValidated: true,
+    isInboxEnabled: false,
+    acceptedTosVersion: 1,
+    id: "ISPXNB32R82Y766D-0000000000000001",
+    version: 1,
+    _self: "eb28139a-f875-4276-81c5-b3f7b01f712a"
+  },
+  {
+    fiscalCode: "not-valid",
+    isEmailValidated: true,
+    isInboxEnabled: false,
+    acceptedTosVersion: 1,
+    id: "id",
+    version: 0
+  },
+  {
+    email: "Eleanore.Kuphal@example.net",
+    fiscalCode: "VSFNVG14A39Y596X",
+    isEmailValidated: false,
+    isInboxEnabled: false,
+    id: "VSFNVG14A39Y596X-0000000000000001",
+    version: 1,
+    acceptedTosVersion: 0,
+    _self: "aa406556-b003-4387-89ef-8f127ec9b2da"
+  },
+  {
+    email: "Eleanore.Kuphal@example.net",
+    fiscalCode: "VSFNVG14A39Y596X",
+    isEmailValidated: true,
+    isInboxEnabled: false,
+    id: "VSFNVG14A39Y596X-0000000000000002",
+    version: 2,
+    _self: "f43db63b-5549-403d-98a5-e781934c796f"
+  }
+];
+
+jest.spyOn(mockProfileModel, "find").mockImplementation(([id, fiscalCode]) =>
+  pipe(
+    mockProfiles.find(
+      profile => profile.id === id && profile.fiscalCode === fiscalCode
+    ),
+    foundProfile => TE.right(O.fromNullable(foundProfile as RetrievedProfile))
+  )
+);
+
+const mockLogger = {
+  error: jest.fn()
+};
+
+const mockDependencies = {
+  dataTableProfileEmailsRepository: mockDataTableProfileEmailsRepository,
+  profileModel: mockProfileModel,
+  logger: mockLogger
+};
+
+describe("handler function", () => {
+  it("should call get, insert and delete methods with no errors", async () => {
+    const documents = mockProfiles;
+
+    await handler(documents)(mockDependencies)();
+
+    expect(mockProfileModel.find).toHaveBeenCalledTimes(5);
+    expect(mockDataTableProfileEmailsRepository.insert).toHaveBeenCalledTimes(
+      5
+    );
+    expect(mockDataTableProfileEmailsRepository.delete).toHaveBeenCalledTimes(
+      1
+    );
+  });
+
+  it("should call mockLogger.error when error in insertProfileEmail occurs", async () => {
+    const mockDocuments = mockProfiles.filter(
```

Comment:
this filter is prone to errors in my opinion. I would suggest something like the following:
```ts
const take = (id: string, arr: typeof mockProfiles) =>
  pipe(
    arr,
    A.filter(e => e.id === id)
  );

const getSelfProp = (i: Array<Record<string, unknown>>) =>
  pipe(
    i,
    A.head,
    O.chain(e => R.lookup("_self", e)),
    O.toUndefined
  );
```

```suggestion
    const mockDocuments = take(generateId("DRUQIL23A18Y188X", 0), mockProfiles);
```